src/prefect/server/orchestration/core_policy.py

```diff
@@ -273,8 +273,9 @@ class RetryFailedFlows(BaseOrchestrationRule):
     Rejects failed states and schedules a retry if the retry limit has not been reached.
 
     This rule rejects transitions into a failed state if `retries` has been
-    set and the run count has not reached the specified limit. The client will be
-    instructed to transition into a scheduled state to retry flow execution.
+    set, the run count has not reached the specified limit, and the client
+    asserts it is a retriable flow run. The client will be instructed to
+    transition into a scheduled state to retry flow execution.
```

Comment:
My understanding was this PR is only for retriable task runs?

Reply:
It is. The only change here was informing anyone inspecting `RetryFailedFlows` that there are now 3 conditions that have to be met.

Reply:
Ah, I actually meant to refer to line 296 (the 3rd condition). Why would the tasks being retriable matter when we decide to retry a failed flow?

Reply:
Ah, I may have been thinking I will do this for tasks and flows when I started. I can walk that back.

Reply:
Thanks @dominictarro :)